Source/Fuse.Elements/Resources/HttpImageSource.uno

```diff
@@ -178,7 +226,21 @@ namespace Fuse.Resources
 			{
 				try
 				{
-					_tex = TextureLoader.ByteArrayToTexture2DContentType(_data, _contentType);
+					if (_data == null && _contentType == null)
+					{
+						_data = File.ReadAllBytes(_filename);
+						_tex = TextureLoader.ByteArrayToTexture2DFilename(_data, _filename);
```

Comment:
I don't think this is going to work. `_filename` doesn't contain any extension, and will just end up [throwing an exception](https://github.com/fuse-open/fuselibs/blob/9c9ab807f1c6ea0a6a3256940b543d41b8f033fb/Source/Experimental.TextureLoader/TextureLoader.uno#L99)...

Reply:
It will work because of this block statement only be executed by [this statement](https://github.com/fuse-open/fuselibs/pull/1273/files/c51dd1a1c3467cbb66b3bb1eebe359d8a075716f#diff-7c9578067052e57d861466f1e69e34a3R136).  And [that statement](https://github.com/fuse-open/fuselibs/pull/1273/files/c51dd1a1c3467cbb66b3bb1eebe359d8a075716f#diff-7c9578067052e57d861466f1e69e34a3R136) only be called when `FindFileFromUrl` method return true

Reply:
Ah, I see, thanks for clarifying. I don't think this is very clear from the code, though. Perhaps we could use a differently named variable than `_filename`, when it's not actually the full filename? For instance `_filenameBase` or something like that?

I also think this code could use some tests that can be used to verify that corner-cases like this work...

Reply:
(note that the last part is "could"; I don't want to force you to do this if you're pressed on time or something)

Reply:
Yeah I have to refactor some variable and method name to better readability